src/ducks/Watchlists/Widgets/TopBar/BaseActions/Items.js

```diff
@@ -0,0 +1,151 @@
+  <Item {...props} icon='disk' border className={styles.saveAsNonAuthor}>
+    Save as
+  </Item>
+)
+
+export const Trigger = ({
+  type,
+  title,
+  watchlist,
+  forwardedRef,
+  isActive,
+  onPrimaryAction,
+  isLoading,
+  openMenu
+}) => {
+  const { name, description, isPublic } = watchlist
+  const [opened, setOpened] = useState(false)
+
+  const onSubmit = props => onPrimaryAction(props).then(() => setOpened(false))
+
+  return (
+    <div className={styles.trigger} ref={forwardedRef}>
+      <EditForm
+        type={type}
+        open={opened}
+        id={watchlist.id}
+        watchlist={watchlist}
+        isLoading={isLoading}
+        toggleOpen={setOpened}
+        title={'Edit ' + title}
+        onFormSubmit={onSubmit}
+        settings={{ name, description, isPublic }}
+        trigger={<Button className={styles.trigger__text}>Edit</Button>}
+      />
+      <div
+        className={cx(
+          styles.trigger__arrowBtn,
+          isActive && styles.trigger__arrowBtn_active
+        )}
+        onClick={openMenu}
+      >
+        <Icon
+          type='arrow-down'
+          className={cx(
+            styles.trigger__arrow,
+            isActive && styles.trigger__arrow_active
+          )}
+        />
+      </div>
+    </div>
+  )
+}
+
+export const Delete = ({ id, name, title }) => (
+  <DeleteAction
+    title={`Do you want to delete this ${title}?`}
+    id={id}
+    name={name}
+    trigger={
+      <Item icon='remove' className={styles.delete}>
+        Delete
+      </Item>
+    }
+  />
+)
+
+export const New = ({ type, onSubmit }) => (
+  <NewAction
+    type={type}
+    onSubmit={onSubmit}
+    trigger={<Item icon='plus-round'>New</Item>}
+  />
+)
+
+export const SaveAs = ({ type, watchlist }) => {
+  const showDuplicate = ['SCREENER', 'PROJECT', 'BLOCKCHAIN_ADDRESS'].includes(
+    type
+  )
+  const iconName = showDuplicate ? 'duplicate' : 'disk'
+  return (
+    <SaveAsAction
+      type={type}
+      watchlist={watchlist}
+      trigger={
+        <Item icon={iconName} className={iconName}>
+          {showDuplicate ? 'Duplicate' : 'Save as'}
+        </Item>
+      }
+    />
+  )
+}
+
+export const Edit = ({ type, title, watchlist, onSubmit, isLoading }) => {
+  const [opened, setOpened] = useState(false)
+  const { name, description, isPublic } = watchlist
+
+  return (
+    <EditForm
+      type={type}
+      open={opened}
+      id={watchlist.id}
+      watchlist={watchlist}
+      isLoading={isLoading}
+      toggleOpen={setOpened}
+      title={'Edit ' + title}
+      trigger={<Item icon='edit'>Edit</Item>}
+      settings={{ name, description, isPublic }}
+      onFormSubmit={payload => onSubmit(payload).then(() => setOpened(false))}
+    />
+  )
+}
+
+export const Copy = ({ watchlist }) => {
+  return (
+    <CopyAction
+      id={watchlist.id}
+      assets={watchlist.listItems.map(l => l.project)}
+      trigger={<Item icon='copy'>Copy assets</Item>}
+    />
+  )
+}
```

Comment:
Copy of the `src/ducks/Watchlists/Widgets/TopPanel/BaseActions/Items.js` module. Reuse existing components. If they can't be used directly - create an abstraction for a general use.